app/src/main/java/org/xbmc/kore/host/HostConnectionObserver.java

```diff
@@ -160,6 +180,10 @@ public void run() {
             if (!applicationEventsObservers.isEmpty())
                 getApplicationProperties();
 
+            if ( ! playlistEventsObservers.isEmpty() ) {
+                checkPlaylist();
+            }
+
```

Comment:
We should be careful about this. Through HTTP we are polling Kodi every 2 seconds. Up until now, we were calling Kodi 4 times (`getActivePlayers`, `getPlayerProperties`, `getItems` and `getProperties`). With this, we're adding 4 more calls, for a total if 8 calls to Kodi, every 2 seconds.
We should evaluate if this might start to impact Kodi. One possible solution is to poll the playlists less often, in another runnable that runs every 4 seconds or so (that was the initial though, when we had more than one runnable here).

Reply:
People connected to Kodi on WiFi will probably not have any problems. On slower connections this may become an issue with huge playlists. We could just disable this feature for HTTP connections? (I'm not a fan of adding another setting to allow users to enable/disable this.)

Reply:
Why not simply decrease the frequency of this? Creating another runnable might be too much trouble, but you could just check the playlist every 2 runs of this runnable, for instance.